feat: add git lfs admin option (#10748)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

## allowCustomCrateRegistries

## allowGitLfs

> **Note** You need to also configure `gitNoVerify` to allow commit and push git lfs objects.

Reviewer: I feel like this text is missing something?

How about:

```suggestion
**Note** You need to also configure `gitNoVerify` to `true` to allow Renovate to commit and push Git LFS objects.
```

I think the correct way to capitalize is `Git LFS` by the way. 😉

Author: Sure, but `gitNoVerify` is a string array option, so no `true` here

Reviewer: Updated my suggestion. 😄

## allowCustomCrateRegistries

## allowGitLfs

> **Note** You need to also set the Git commands that Renovate passes to the `gitNoVerify` option.

Reviewer 2: This isn't fully correct. Renovate isn't passing any commands to the `giNoVerify` option. Instead, I would perhaps phrase it something like
> You also need to set the `gitNoVerify` option to the appropriate values for LFS to work.
...ish.

> **Note** You need to also set the Git commands that Renovate passes to the `gitNoVerify` option.
> This way Renovate can commit and push Git LFS objects.

Reviewer: ```suggestion
Use this option to allow Renovate to work with repositories which use Git LFS (Git Large File Support).

**Note** You need to also set the Git commands that Renovate passes to the `gitNoVerify` option.
This way Renovate can commit and push Git LFS objects.
```

How about a short introductory sentence, that explains what this option does?

Or do we think the section name is good enough on its own?

Author: Remember, that the docs compiler will also prepend the message and default values from config definitions file.

## allowCustomCrateRegistries

## allowGitLfs

**Note** You need to also set the Git commands that Renovate passes to the `gitNoVerify` option.
This way Renovate can commit and push Git LFS objects.

> **Note** This currently don't work in official renovate docker images, see [containerbase/buildpack#13](https://github.com/containerbase/buildpack/issues/13)

Reviewer: ```suggestion
**Note** `allowGitLfs` does not work in the official Renovate docker images, see issue [containerbase/buildpack#13](https://github.com/containerbase/buildpack/issues/13) for more information.
```

Read the documentation for [git commit --no-verify](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---no-verify) and [git push --no-verify](https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-verify) to learn exactly what each flag does.
To learn more about Git hooks, read the [Pro Git 2 book, section on Git Hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks).

> **Note** If you use git-lfs checkout `allowGitLfs` option.

Reviewer: ```suggestion
**Note** If you use Git LFS you also need to configure the `allowGitLfs` option to `true`.
```

Maybe something like this is clearer?

Read the documentation for [git commit --no-verify](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---no-verify) and [git push --no-verify](https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-verify) to learn exactly what each flag does.
To learn more about Git hooks, read the [Pro Git 2 book, section on Git Hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks).

> **Note** If you use Git LFS you also need to configure the `allowGitLfs` option to `true`.

Reviewer: ```suggestion
**Note** If you use Git LFS you also need to set the `allowGitLfs` option to `true`.
```

Maybe `set` is a better word than `configure` here?
